src/rgw/rgw_basic_types.h

```diff
@@ -31,6 +31,7 @@
 #include "rgw_user_types.h"
 #include "rgw_bucket_types.h"
 #include "rgw_obj_types.h"
+#include "rgw_cksum.h"
```

Comment:
rgw_cksum.h is all header-only, so this pulls a bunch of header dependencies into rgw_basic_types.h that aren't needed here. all that `RGWUploadPartInfo` really needs is a struct containing the Type enum, the digest value array, and their encodings

could we separate out a rgw_cksum_types.h that only has the essentials?

Reply:
i remember seeing some changes related to this, but they've since disappeared. what happened to rgw_cksum_types.h?

Reply:
oh, yeah.  I thought that was still present.  I'll look through my snapshot branches

Reply:
(I rebased a bunch of times)